setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+                # this is a file
+                return os.path.join(output_data_dir, maybe_file)  # output_data_dir should already be a relative path dir to the repo TLD
+            else:
+                return maybe_file
+
+        for k, v in json.loads(input_json.strip()).items():
+            input_json_dict[if_file_convert(k)] = recursive_json_apply(v, if_file_convert)
+
+    config_entry["inputs"] = {
+        "dir": wdl_dir,
+        "wdl": wdl_base,
+        "json_string": input_json_dict
+    }
+
+    if output_json is not None:
+        config_entry["outputs"] = json.loads(output_json.strip())
+    else:
+        config_entry["outputs"] = {}
+
+    output_var_types = extract_output_types(wdl_file, bool(is_fail))
+
+    if "return_code" not in config_entry:
+        config_entry["return_code"] = "*"
+    elif isinstance(config_entry["return_code"], str):
+        config_entry["return_code"] = [config_entry["return_code"]]
+
+    if bool(is_fail):
+        config_entry["outputs"] = {}
+    else:
+        if output_json is not None:
+            target_output_metadata = get_not_none(target_metadata, "outputs")
+            for k, v in json.loads(output_json.strip()).items():
+                k_base = k.split(".")[-1]
+                output_type = output_var_types[k_base]
+                config_entry["outputs"][k] = {
+                    "type": output_type,
+                    "value": convert_typed_output_values_from_string(v, output_type, data_dir, get_not_none(get_not_none(target_output_metadata, k), "value"))
+                }
+
+    if "type" not in config_entry:
+        config_entry["type"] = "task" if is_task else "workflow"
+    if "target" not in config_entry:
+        config_entry["target"] = target
+    if "priority" not in config_entry:
+        config_entry["priority"] = "required"
+    if "exclude_output" not in config_entry:
+        config_entry["exclude_output"] = []
+    elif isinstance(config_entry["exclude_output"], str):
+        config_entry["exclude_output"] = [config_entry["exclude_output"]]
+    if "dependencies" not in config_entry:
+        config_entry["dependencies"] = []
+    elif isinstance(config_entry["dependencies"], str):
+        config_entry["dependencies"] = [config_entry["dependencies"]]
+    # add metadata dependencies
+    if get_not_none(target_metadata, "dependencies") is not None:
+        config_entry["dependencies"].extend(target_metadata["dependencies"])
+    if "tags" not in config_entry:
+        config_entry["tags"] = []
+    elif isinstance(config_entry["tags"], str):
+        config_entry["tags"] = [config_entry["tags"]]
+
+    config_entry["tags"].append("unit")
+
+    config_entry["description"] = f"Unit test for {file_name}"
+
+    config_entry["versions"] = [version]
+
+    config.append(config_entry)
+
+
+def write_test_files(m: re.Match, output_dir: Path, version: str):
+    """
+    Given the parsed regex, write the test file into the output directory.
```

Comment:
I think a parsed regex would be the non-str object `re` uses for compiled regexes, not `re.Match`.

Reply:
Fixed the docstring